src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -301,73 +344,85 @@ export class DataSourceSavedObjectsClientWrapper {
+        encryptionContext = { endpoint };
+        break;
       case AuthType.UsernamePasswordType:
         const { credentials } = auth;
         if (!credentials) {
           throw SavedObjectsErrorHelpers.createBadRequestError(
-            'Update failed due to deprecated data source: "credentials" missing. Please delete and create another data source.'
+            'Failed to update existing data source: "credentials" missing. Please delete and create another data source.'
           );
         }
 
         const { username, password } = credentials;
 
         if (!username) {
           throw SavedObjectsErrorHelpers.createBadRequestError(
-            'Update failed due to deprecated data source: "auth.credentials.username" missing. Please delete and create another data source.'
+            'Failed to update existing data source: "auth.credentials.username" missing. Please delete and create another data source.'
           );
         }
 
         if (!password) {
           throw SavedObjectsErrorHelpers.createBadRequestError(
-            'Update failed due to deprecated data source: "auth.credentials.username" missing. Please delete and create another data source.'
-          );
-        }
-
-        const { encryptionContext } = await this.cryptography
-          .decodeAndDecrypt(password)
-          .catch((err: any) => {
-            const errMsg = `Failed to update existing data source for dataSourceId [${id}]: unable to decrypt "auth.credentials.password"`;
-            this.logger.error(errMsg);
-            this.logger.error(err);
-            throw SavedObjectsErrorHelpers.decorateBadRequestError(err, errMsg);
-          });
-
-        if (encryptionContext.endpoint !== endpoint) {
-          throw SavedObjectsErrorHelpers.createBadRequestError(
-            'Update failed due to deprecated data source: "endpoint" contaminated. Please delete and create another data source.'
+            'Failed to update existing data source: "auth.credentials.username" missing. Please delete and create another data source.'
           );
         }
-        return encryptionContext;
+        encryptionContext = await this.getEncryptionContext(password);
+        break;
+      case AuthType.SigV4:
+        const { accessKey } = auth.credentials as SigV4Content;
+        encryptionContext = await this.getEncryptionContext(accessKey);
+        break;
       default:
-        throw SavedObjectsErrorHelpers.createBadRequestError(`Invalid auth type: '${type}'`);
+        throw SavedObjectsErrorHelpers.createBadRequestError(`Invalid auth type: '${auth.type}'`);
+    }
+
+    // validate encryption context
+    if (encryptionContext.endpoint !== endpoint) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(
+        'Failed to update existing data source: "endpoint" contaminated. Please delete and create another data source.'
+      );
     }
+    return encryptionContext;
+  }
+
+  private async getEncryptionContext(input: string) {
```

Comment:
let's rename this function as `getEncryptionContextFromCipher` and name the parameter name to be `cipherText` to make it more readable